base/compiler/typelimits.jl

```diff
@@ -227,7 +233,12 @@ function type_more_complex(@nospecialize(t), @nospecialize(c), sources::SimpleVe
     end
     # rules for various comparison types
     if isa(c, TypeVar)
-        tupledepth = 1 # allow replacing a TypeVar with a concrete value (since we know the UnionAll must be in covariant position)
+        if tupledepth == 0
+            # Even if we're out of tupledepth, allow replacing a TypeVar with a concrete value (since we know the UnionAll must be in covariant position).
```

Comment:
```suggestion
            # Even if we're out of tupledepth, allow replacing a TypeVar with a concrete value
```

This part of the comment was not really true, since we don't actually know that this UnionAll was bound at `tupledepth==1` (it might instead have been bound inside a parameter, such as in `Vector{Vector{T} where T}`)